synapse/metrics/__init__.py

```diff
@@ -158,18 +158,33 @@ def runUntilCurrentTimer(func):
 
     @functools.wraps(func)
     def f(*args, **kwargs):
-        pending_calls = len(reactor.getDelayedCalls())
+        now = reactor.seconds()
+        num_pending = 0
+
+        # _newTimedCalls is one long list of *all* pending calls. Below loop
+        # is based off of impl of reactor.runUntilCurrent
+        for p in reactor._newTimedCalls:
```

Comment:
We might want to check that the _newTimedCalls attribute exists if we are rummaging around in twisted internals.